### PR TITLE
Add logs of fetching keycloak authorization code, and fix pr 921 comm…

### PR DESCRIPTION
**Pull request by nessiLahav**
#### What does this PR do?
Add logs of fetching Keycloak authorization code in testing,
Add add fixes according to comments in: #921 
#### Link to build in Jenkins (if appropriate)
https://jenkins.conjur.net/job/cyberark--conjur/job/pr_921_fixes/1/
#### Has the Version and Changelog been updated?

> Can we make a blog post, video, or animated GIF of this?

> Has this change been documented (Readme, docs, etc.)?

> Does the knowledge base need an update?

### CODE REVIEW

File: ci/authn-oidc/phantomjs/keycloak_login.js [hunk 1]
```diff
@@ -8,38 +8,42 @@ var webPage = require('webpage');
...
+log('Date in container: ' + date);
 page.settings.userAgent = 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36';
 page.settings.javascriptEnabled = true;
 page.settings.loadImages = false;//Script is much faster with this field set to false
 phantom.cookiesEnabled = true;
 phantom.javascriptEnabled = true;
 /*********SETTINGS END*****************/
 
-console.log('All settings loaded, start with execution');
+log('All settings loaded, start with execution');
```
Comment (orenbm) [inline ci/authn-oidc/phantomjs/keycloak_login.js]: maybe write what settings were loaded?

File: ci/authn-oidc/phantomjs/keycloak_login.js [hunk 2]
```diff
@@ -48,14 +52,13 @@ steps = [
     },
 
     function(){
-		     console.log("Wait keycloak to login user");
-         var fs = require('fs');
+		     log("Wait keycloak to login user");
```
Comment (orenbm) [inline ci/authn-oidc/phantomjs/keycloak_login.js]: "wait for keycloak..."

File: dev/docker-compose.yml [hunk 3]
```diff
@@ -113,8 +113,8 @@ services:
 
   oidc-phantomjs:
     environment:
-      - KEYCLOAK_USER=alice
-      - KEYCLOAK_PASSWORD=alice
+      - KEYCLOAK_APP_USER=alice
+      - KEYCLOAK_APP_USER_PASSWORD=alice
```
Comment (orenbm) [inline dev/docker-compose.yml]: does this work? I think that `KEYCLOAK_USER` & `KEYCLOAK_PASSWORD` are required for setting the adminsitrator user in keycloak.

Please verify that the tests pass in the `./cli exec` flow and not only `ci/test`

Comment (nessiLahav) [reply dev/docker-compose.yml]: I test them both and they are working please double check me if you are thinking that i am missing something

File: ci/authn-oidc/phantomjs/keycloak_login.js (hunk 1 above)
Comment (nessiLahav) [reply ci/authn-oidc/phantomjs/keycloak_login.js]: It is not really interesting, this is only an help script to fetch the authorization code

File: dev/docker-compose.yml (hunk 3 above)
Comment (orenbm) [reply dev/docker-compose.yml]: sorry i thought these were params of the `oidc-keycloak` container

File: ci/authn-oidc/phantomjs/keycloak_login.js [hunk 4]
```diff
@@ -8,38 +8,42 @@ var webPage = require('webpage');
...
 phantom.javascriptEnabled = true;
 /*********SETTINGS END*****************/
 
-console.log('All settings loaded, start with execution');
+log('All settings loaded, start with execution');
 page.onConsoleMessage = function(msg) {
-    console.log(msg);
+    log(msg);
 };
 /**********DEFINE STEPS THAT FANTOM SHOULD DO***********************/
```
Comment (jtuttle) [inline ci/authn-oidc/phantomjs/keycloak_login.js]: typo: `FANTOM` should be `PHANTOM`.

File: ci/authn-oidc/phantomjs/keycloak_login.js [hunk 5]
```diff
@@ -48,14 +52,13 @@ steps = [
     },
 
     function(){
-		     console.log("Wait keycloak to login user");
-         var fs = require('fs');
+		     log("Wait for keycloak to login user");
```
Comment (jtuttle) [inline ci/authn-oidc/phantomjs/keycloak_login.js]: Can you please fix the indenting here? (And everywhere else in the file that it's slightly off).

File: ci/authn-oidc/phantomjs/keycloak_login.js [hunk 6]
```diff
@@ -66,28 +69,33 @@ interval = setInterval(executeRequestsStepByStep,50);
...
-        console.log("test complete!");
+        log("test complete!");
         phantom.exit();
     }
 }
 
+function log(msg) {
+    console.log(msg);
+    fs.write(logfile,msg,'a');
+
```
Comment (jtuttle) [inline ci/authn-oidc/phantomjs/keycloak_login.js]: Remove blank line.

File: cucumber/authenticators/features/support/authn_oidc_helper.rb [hunk 7]
```diff
@@ -76,7 +74,7 @@ def oidc_redirect_uri
   end
 
   def oidc_auth_code
-    raise 'Authorization code is not initialized' if @oidc_auth_code.blank?
+    raise 'Authorization code is not initialized, Additional logs exists in keycloak_login.[date].log file' if @oidc_auth_code.blank?
```
Comment (jtuttle) [inline cucumber/authenticators/features/support/authn_oidc_helper.rb]: Please break this out over multiple lines.

**Pull request merged**